### PR TITLE
[Merged by Bors] - [cu-2dvdhu2] Phase 1: Documenting Existing Oracle Design for Tracking Prices

### PR DESCRIPTION
**Pull request by vimukthi-git**
## Description
This is a post-facto design doc for the Apollo oracle.

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2dvdhu2 Oracle design doc](https://app.clickup.com/t/2dvdhu2)

File: frame/oracle/design/design.adoc [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  if (1. is_requested(asset_id)) then (yes)
+    :2. fetch_price_and_send_signed(asset_id, asset_info);
+  endif
+endwhile
+stop
+----
+
+Following are the sub-procedures from above main algorithm,
+
+.Offchain Woker: is_requested
```
Comment (vimukthi-git) [inline frame/oracle/design/design.adoc:245]:
```suggestion
.Offchain Worker: is_requested
```

File: frame/oracle/design/design.adoc [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+    < current_block;
+    note left: Price is requested if\nthe last update is too old
+    stop
+else (no)
+    :return false;
+    stop
+endif
+----
+
+.Offchain Woker: fetch_price_and_send_signed
```
Comment (vimukthi-git) [inline frame/oracle/design/design.adoc:265]:
```suggestion
.Offchain Worker: fetch_price_and_send_signed
```

**Comment by vimukthi-git** [comment]
Note to reviewers: This phase 1 only captures only the most important parts of the oracle design such as offchain worker, on_initialization hook, and submit_price extrinsic. We would work on adding more details as we get more time as a team. Also this does not make any proposals to update the design, it is just documenting the existing design.

**Review by vivekvpandya — APPROVED** [review]

File: frame/oracle/design/design.adoc [hunk 3]
```diff
@@ -0,0 +1,695 @@
...
+cloud {
+    node ".. Price-Source n" <<price-source>> as n {
+    }
+}
+
+"Oracle Provider" as op
+op -up-> as
+----
+
+NOTE: It is proposed to rename the existing pallet-oracle as pallet-apollo here.
```
Comment (benluelo) [inline frame/oracle/design/design.adoc:168]: Can we rename vault to cubic while we're at it :laughing:

Comment (vimukthi-git) [reply frame/oracle/design/design.adoc:168]: haha sure we can, but not something relevant to this PR :)

**Comment by vimukthi-git** [comment]
As per my discussion with @KaiserKarel I will update the future improvements section here to include rewriting the price server in typescript to be easily extensible to add new price feeds.

**Review by KaiserKarel — APPROVED** [review]

**Review by vivekvpandya — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

**Pull request closed without merging**